source/plugins/go/src/oms.go

```diff
@@ -1149,6 +1152,141 @@ func UpdateNumTelegrafMetricsSentTelemetry(numMetricsSent int, numSendErrors int
 	ContainerLogTelemetryMutex.Unlock()
 }
 
+func processIncludes(kubernetesMetadataMap map[string]interface{}, includesList []string) map[string]interface{} {
+	includedMetadata := make(map[string]interface{})
+
+	// pre process image related fields
+	var imageRepo, imageName, imageTag, imageID string
+	imageProcessed := false // Flag to check if image processing is required
+	for _, include := range includesList {
+		if include == "imageid" || include == "imagerepo" || include == "image" || include == "imagetag" {
+			imageProcessed = true
+			break
+		}
+	}
+
+	if imageProcessed {
+		if hash, ok := kubernetesMetadataMap["container_hash"].(string); ok {
+			imageID = extractImageID(hash)
+		}
+		if image, ok := kubernetesMetadataMap["container_image"].(string); ok {
+			imageRepo, imageName, imageTag = parseImageDetails(image)
+		}
+	}
+
+	for _, include := range includesList {
+		switch include {
+		case "poduid":
+			if val, ok := kubernetesMetadataMap["pod_id"]; ok {
+				includedMetadata["podUid"] = val
+			}
+		case "podlabels":
+			if val, ok := kubernetesMetadataMap["labels"]; ok {
+				includedMetadata["podLabels"] = val
```

Comment:
Did you test what max labels size can be ingested? is it 64KB? if yes, then we should cap up to  64KB in the agent  and then discard more than 64kB.

Reply:
A kubernete label key must be no longer than 63 characters